### PR TITLE
[RDY] man.vim: Set iskeyword explicitly.

### PR DESCRIPTION
**Pull request by shade-of-noon**
This also fixes `:Man!`, which wasn't setting 'iskeyword' to contain
parentheses.

Tagging the maintainer, @nhooyr.

### CODE REVIEW

File: runtime/autoload/man.vim [hunk 1]
```diff
@@ -140,7 +140,6 @@ function! s:put_page(page) abort
   setlocal noreadonly
   setlocal noswapfile
   " git-ls-files(1) is all one keyword/tag-target
```
Comment (nhooyr) [inline runtime/autoload/man.vim:142]: Comment needs to be moved.

**Review by nhooyr — COMMENTED** [review]
Great catch @shade-of-noon

File: runtime/autoload/man.vim (hunk 1 above)
Comment (shade-of-noon) [reply runtime/autoload/man.vim]: Done.

**Review by shade-of-noon — COMMENTED** [review]
Fixed the misplaced comment.

**Review by nhooyr — APPROVED** [review]
Awesome! 🚀

**Comment by shade-of-noon** [comment]
Hello. I added a mapping so that double clicking jumps to the man page of the word under the cursor.

**Comment by nhooyr** [comment]
Not sure if we should have such a mapping. It might be better to add a default mapping that does that with `tagfunc`.

**Comment by nhooyr** [comment]
Or perhaps with `'keywordprg'`

**Comment by shade-of-noon** [comment]
Fixed a typo (wrote Map instead of Man). This idea came to me when I so that the help buffers had this mapping.

> Not sure if we should have such a mapping. It might be better to add a default mapping that does that with tagfunc.

Can you elaborate? I don't quite understand what you mean…

File: runtime/ftplugin/man.vim [hunk 2]
```diff
@@ -31,7 +31,7 @@ if !exists('g:no_plugin_maps') && !exists('g:no_man_maps')
   nnoremap <silent> <buffer> j             gj
   nnoremap <silent> <buffer> k             gk
   nnoremap <silent> <buffer> gO            :call man#show_toc()<CR>
-  nnoremap <silent> <buffer> <2-LeftMouse> :Map<CR>
+  nnoremap <silent> <buffer> <2-LeftMouse> :Man<CR>
```
Comment (nhooyr) [inline runtime/ftplugin/man.vim:34]: Let's squash this into the previous commit.

**Review by nhooyr — APPROVED** [review]

**Comment by nhooyr** [comment]
> Can you elaborate? I don't quite understand what you mean…

When you press `ctrl-]`, `tagfunc` is invoked to take you to the tag (manpage) under the cursor. I'm saying why don't we add double click as an additional keybind for `tagfunc` but we can do that later.

File: runtime/ftplugin/man.vim (hunk 2 above)
Comment (shade-of-noon) [reply runtime/ftplugin/man.vim:34]: Done!

**Comment by shade-of-noon** [comment]
> When you press `ctrl-]`, `tagfunc` is invoked to take you to the tag (manpage) under the cursor. I'm saying why don't we add double click as an additional keybind for `tagfunc` but we can do that later.

Unfortunately, in its current state, `:Man!` (nvim as man pager) doesn't generate the tags. Also, the tags generated (by `:Man`) only work for pages which start with the same prefix (e.g., man, mandb, man-pages, etc.).

For example, if you run `man -P 'nvim +Man!' man` since the tags are not generated, you won't be able to jump to `man(7)` (a different section); and since the tags only match words with a common prefix, you won't be able to jump to `groff(7)`  by using `ctrl+]`.

**Comment by nhooyr** [comment]
> Unfortunately, in its current state, the tags generated (by :Man) only work for pages which start with the same prefix (e.g., man, mandb, man-pages, etc.).

Not sure why you're seeing that behaviour. `<C-]>` on `builtin` in `printf(1)` works perfectly for me.

**Comment by nhooyr** [comment]
At least, it works as expected for most people otherwise we'd have seen many bug reports.

**Comment by shade-of-noon** [comment]
That's weird. I'm just test with vanilla Neovim from master:
1. Using Neovim as man pager doesn't produce tags.
2. The tag function only generates tags for pages with a common prefix and a common section.

I checked with
- `man -P 'nvim -u NORC +Man!'` ---> no tags (`:tselect` produces an empty stack error).
- From within Neovim, `:Man man` ---> only tags for pages whose names start with `man` in section 1.

Are you sure you don't have `CTRL-]` customized? A plugin or something like that?

**Comment by nhooyr** [comment]
From `:help :tselect`
> When [name] is not given, the last tag name from the tag stack is used.

**Comment by nhooyr** [comment]
E.g. try `:tselect .`

**Comment by nhooyr** [comment]
Whoops nvm, i'm not really sure what you'd use to list all tags in a file.

**Comment by nhooyr** [comment]
This is ready for merge @jamessan

**Comment by nhooyr** [comment]
There is that openbsd job failure but I'm certain it's unrelated.

The trace indicates its in `test/functional/ui/spell_spec.lua:26: FIXME #12104`

**Comment by nhooyr** [comment]
Thanks again @shade-of-noon <3

**Pull request merged**